### PR TITLE
Shrink lambdas using docker

### PR DESCRIPTION
**Pull request by ividito**
Lambdas were not deployable due to their package size (numpy and pandas are needed for zarr ingests). This PR refactors our lambdas to use Docker instead. I had to reorganize imports, as well as adjust tests and local dev instructions to align with the new structure.

### CODE REVIEW

**Comment by ividito** [comment]
Tests are failing due to modified import paths, but the dockerized lambdas seem to be working - still need to test some different paths through the application. I'll be rewriting the tests to match the new format shortly. This branch is currently deployed on the dev environment.

File: cdk/stack.py [hunk 1]
```diff
@@ -240,20 +241,24 @@ def build_api_lambda(
...
             timeout=Duration.seconds(30),
+            handler="handler.handler",
             role=handler_role,
             environment={"DB_SECRET_ARN": db_secret.secret_arn, **env},
             vpc=db_vpc,
             vpc_subnets=ec2.SubnetSelection(
                 subnet_type=ec2.SubnetType.PUBLIC
                 if db_subnet_public
-                else ec2.SubnetType.PRIVATE_WITH_NAT
+                else ec2.SubnetType.PRIVATE_ISOLATED
```
Comment (ividito) [inline cdk/stack.py:261]: This addresses a different deployment issue - `PRIVATE_ISOLATED` is the subnet type we currently use in the `veda-backend` VPC.

**Review by moradology — APPROVED** [review]
I'm not sure I understand the implications of `PRIVATE_WITH_NAT` vs `PRIVATE_ISOLATED`. Otherwise, things look clean and correct

**Review by smohiudd — APPROVED** [review]
Looks good. Ran `pytest` and `cdk synth` with no issues.

**Pull request merged**